plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/thrift/ThriftHiveMetastoreClient.java

```diff
@@ -220,12 +222,22 @@ public Table getTableWithCapabilities(String databaseName, String tableName)
     }
 
     @Override
-    public void setTableColumnStatistics(String databaseName, String tableName, List<ColumnStatisticsObj> statistics)
+    public void setTableColumnStatistics(String databaseName, String tableName, List<ColumnStatisticsObj> statistics, AcidTransaction transaction)
             throws TException
     {
         ColumnStatisticsDesc statisticsDescription = new ColumnStatisticsDesc(true, databaseName, tableName);
-        ColumnStatistics request = new ColumnStatistics(statisticsDescription, statistics);
-        client.update_table_column_statistics(request);
+        ColumnStatistics columnStatistics = new ColumnStatistics(statisticsDescription, statistics);
+        if (transaction.isAnalyze()) {
+            SetPartitionsStatsRequest request = new SetPartitionsStatsRequest();
+            request.setColStats(ImmutableList.of(columnStatistics));
+            request.setNeedMerge(false);
+            request.setWriteId(transaction.getWriteId());
+            request.setValidWriteIdList(getTableWriteIds(databaseName, tableName, transaction.getAcidTransactionId()));
```

Comment:
Explain why it is OK to get valid write ids here, or add

```
// TODO valid write ids should probably be calculated at the beginning of stats collection
```

Reply:
As explained in https://github.com/trinodb/trino/pull/7010#issuecomment-792810890, I couldn't get it to work, and fetching for each partition matches Hive's behavior.  I'll add a comment explaining the result.